Mathlib/RingTheory/Algebraic/MvPolynomial.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import Mathlib.Algebra.MvPolynomial.Supported
+import Mathlib.RingTheory.Algebraic.Basic
+
+/-!
+# Transcendental elements in `MvPolynomial`
+
+This file lists some results on some elements in `MvPolynomial σ R` being transcendental
+over the base ring `R` and subrings `MvPolynomial.supported` of `MvPolynomial σ R`.
+-/
+
+universe u v w
+
+open Polynomial
+
+namespace MvPolynomial
+
+variable {σ : Type*} (R : Type*) [CommRing R]
+
+-- TODO: move to suitable place
+private theorem rename_polynomial_aeval_X
+    {σ τ R : Type*} [CommSemiring R] (f : σ → τ) (i : σ) (p : R[X]) :
+    rename f (Polynomial.aeval (X i) p) = Polynomial.aeval (X (f i) : MvPolynomial τ R) p := by
+  rw [← AlgHom.comp_apply]
+  congr 1; ext1; simp
+
+theorem transcendental_supported_polynomial_aeval_X {i : σ} {s : Set σ} (h : i ∉ s)
+    {f : R[X]} (hf : Transcendental R f) :
+    Transcendental (supported R s) (Polynomial.aeval (X i : MvPolynomial σ R) f) := by
+  classical
+  rw [transcendental_iff_injective] at hf ⊢
+  let g := MvPolynomial.mapAlgHom (R := R) (σ := s) (Polynomial.aeval (R := R) f)
+  replace hf : Function.Injective g := MvPolynomial.map_injective _ hf
+  let u := (Subalgebra.val _).comp
+    ((optionEquivRight R s).symm |>.trans
+      (renameEquiv R (Set.subtypeInsertEquivOption h).symm) |>.trans
+      (supportedEquivMvPolynomial _).symm).toAlgHom |>.comp
+    g |>.comp
+    ((optionEquivLeft R s).symm.trans (optionEquivRight R s)).toAlgHom
+  let v := ((Polynomial.aeval (R := supported R s)
+    (Polynomial.aeval (X i : MvPolynomial σ R) f)).restrictScalars R).comp
+      (Polynomial.mapAlgEquiv (supportedEquivMvPolynomial s).symm).toAlgHom
+  replace hf : Function.Injective u := by
+    simp only [AlgEquiv.toAlgHom_eq_coe, AlgHom.coe_comp, Subalgebra.coe_val,
+      AlgHom.coe_coe, AlgEquiv.coe_trans, Function.comp_assoc, u]
+    apply Subtype.val_injective.comp
+    simp only [EquivLike.comp_injective]
+    apply hf.comp
+    simp only [EquivLike.comp_injective, EquivLike.injective]
+  have h1 : Polynomial.aeval (X i : MvPolynomial σ R) = ((Subalgebra.val _).comp
+      (supportedEquivMvPolynomial _).symm.toAlgHom |>.comp
+      (Polynomial.aeval (X ⟨i, s.mem_insert i⟩ : MvPolynomial ↑(insert i s) R))) := by
+    ext1; simp
+  have h2 : u = v := by
+    simp only [u, v, g]
+    ext1
+    · ext1
+      simp [Set.subtypeInsertEquivOption, Subalgebra.algebraMap_eq]
+    · simp [Set.subtypeInsertEquivOption, rename_polynomial_aeval_X, h1]
```

Comment:
I've successfully inlined the proof of `rename_polynomial_aeval_X` into this:

```suggestion
    · simp [Set.subtypeInsertEquivOption, ← AlgHom.comp_apply (rename _),
        ← Polynomial.aeval_algHom (rename _), h1]
```

and `rename_polynomial_aeval_X` can be removed safely. The `(rename _)` in the proof is required, otherwise it will be `simp` too much.